dist/fabric.js

```diff
@@ -12606,6 +12622,13 @@ fabric.PatternBrush = fabric.util.createClass(fabric.PencilBrush, /** @lends fab
         this._activeObject.clearContextTop();
       }
       fabric.StaticCanvas.prototype.setViewportTransform.call(this, vpt);
+    },
+
+    // Added by aha. Checks if an object is at least partially trasnparent, in
+    // order to prioritize targets behind this object (.e.g, text inside and
+    // behind a slightly transparent square)
+    _isMaybeTransparent: function (object) {
+      return object.fill === 'transparent' || object.opacity < 1;
```

Comment:
Should this include `fill === null` too? I know we set it to 'transparent' but if it were somehow null that'd render the same as transparent?

```suggestion
      return !object.fill || object.fill === 'transparent' || object.opacity < 1;
```

Reply:
Certain objects have `fill: null` that wouldn't be considered transparent, e.g. `SpriteIcon`.

I think `null` is generally used to indicate that the property is no longer used or not being used by an object, so I don't know if we want to key against it in this case.